### PR TITLE
Don't filter out same named NodeDescriptions

### PR DESCRIPTION
**Pull request by dhuebner**
Replaces a name based duplicates filter with URI based one.

See #1176

### CODE REVIEW

File: packages/langium/src/lsp/completion/completion-provider.ts [hunk 1]
```diff
@@ -400,9 +400,10 @@ export class DefaultCompletionProvider implements CompletionProvider {
                 const scope = this.scopeProvider.getScope(refInfo);
                 const duplicateStore = new Set<string>();
                 scope.getAllElements().forEach(e => {
-                    if (!duplicateStore.has(e.name) && this.filterCrossReference(e)) {
+                    const storeId = e.documentUri + e.path;
```
Comment (spoenemann) [inline packages/langium/src/lsp/completion/completion-provider.ts:403]: `${e.documentUri}#${e.path}`

**Review by spoenemann — COMMENTED** [review]
Sometimes one element is in the scope with multiple names. Shall we use a combination of URI, path and name?

File: packages/langium/src/lsp/completion/completion-provider.ts (hunk 1 above)
Comment (dhuebner) [reply packages/langium/src/lsp/completion/completion-provider.ts:403]: Fragment path starts with an `/` which is already a good separator as doc URI normally ends with a file extension I thought. That why I left out an additional separator, but yes we can add `#`.

**Comment by dhuebner** [comment]
> Sometimes one element is in the scope with multiple names. Shall we use a combination of URI, path and name?

I think it is really about to filter out the same AstNodeDescriptions (probably performance reasons), means filter out descriptions that points to the same element using URI + Element path. 
I would rather completely remove this `duplicateStore` as we have an additional method for filtering and completion items are filtered again later on.  See linked issue.

**Comment by msujew** [comment]
@dhuebner I agree. The duplicateStore was supposed to reduce the amount of duplicate suggestions. We have a better method of deduplicating elements now, so we should remove the store.

**Comment by dhuebner** [comment]
@spoenemann 
I removed the `duplicateStore`, please re-view again.

**Review by spoenemann — APPROVED** [review]
👍

**Pull request merged**